### PR TITLE
support for dumping AST of very basic function

### PR DESCRIPTION
**Pull request by colleend**
try `semgrep-core/_build/default/cli/Main.exe -dump_ast tests/kotlin/small-program.kt`. This should give you a basic AST. 

It gives me: 

```
Pr(
  [DefStmt(
     ({name=EId(("foo", ())); attrs=[]; tparams=[];
       info={id_resolved=Ref(None); id_type=Ref(None);
             id_const_literal=Ref(None); };
       },
      FuncDef(
        {fkind=(Function, ()); fparams=[]; frettype=None;
         fbody=Block([ExprStmt(L(Int(("2", ()))), ())]); })))])
```

### CODE REVIEW

**Comment by aryx** [comment]
What's up with the test failures?

File: semgrep-core/parsing/Parse_kotlin_tree_sitter.ml [hunk 1]
```diff
@@ -1358,12 +1366,13 @@ and primary_constructor (env : env) ((v1, v2) : CST.primary_constructor) =
...
   (match x with
-  | `Paren_exp x -> let _ = parenthesized_expression env x in
-      raise Todo
-  | `Simple_id x -> simple_identifier env x
-  | `Lit_cst x -> literal_constant env x
+  | `Paren_exp x -> parenthesized_expression env x
+  | `Simple_id x -> 
+      let id = simple_identifier env x in
+      Id(id, empty_id_info())
+  | `Lit_cst x -> pr2 "in lit_cst"; L (literal_constant env x)
```
Comment (aryx) [inline semgrep-core/parsing/Parse_kotlin_tree_sitter.ml]: there's a pr2 here.
We actually should have a semgrep rule preventing to commit such code.

File: semgrep-core/parsing/Parse_kotlin_tree_sitter.ml [hunk 2]
```diff
@@ -1404,7 +1413,7 @@ and primary_expression (env : env) (x : CST.primary_expression) : AST.expr =
       in
       let v4 = token env v4 (* "]" *) in
       todo env (v1, v2, v3, v4)
-  | `This_exp tok -> let _ = token env tok in raise Todo (* "this" *)
+  | `This_exp tok -> Id(str env tok, empty_id_info()) (* "this" *)
```
Comment (aryx) [inline semgrep-core/parsing/Parse_kotlin_tree_sitter.ml]: There is a IdSpecial for This and Super.
See what I do in Parse_java_tree_sitter.ml

**Review by aryx — APPROVED** [review]
Woohoo! This is great.

File: semgrep-core/parsing/Parse_kotlin_tree_sitter.ml (hunk 1 above)
Comment (colleend) [reply semgrep-core/parsing/Parse_kotlin_tree_sitter.ml]: oh shoot, I'll change that! and very true we should have a semgrep rule for this

**Comment by colleend** [comment]
> What's up with the test failures?

I honestly have no idea why the tests are failing 😓 , but ill look into it 😱 . Seems like the curll to update-docker is not working, but seems like its been a trend for the past few PRs..

File: semgrep-core/parsing/Parse_kotlin_tree_sitter.ml (hunk 1 above)
Comment (colleend) [reply semgrep-core/parsing/Parse_kotlin_tree_sitter.ml]: I will write this this week or next 🤔

**Pull request merged**